[GLUTEN-3559][VL] Enable and fix few failing tests with spark 3.4

ayushi-agarwal opened the pull request:
## What changes were proposed in this pull request?

Fixes following suites:
GlutenSQLQuerySuite
GlutenDataSourceV2Suite
GlutenQueryExecutionErrorsSuite
GlutenCSVSuite
GlutenJsonSuite
GlutenParquetSchemaSuite

(Partially Fixes: \#3559)

## How was this patch tested?

Ran UTs locally

ayushi-agarwal commented:
cc @zhli1142015 @JkSelf

JkSelf commented on gluten-ut/spark34/src/test/scala/io/glutenproject/utils/velox/VeloxTestSettings.scala:
```diff
@@ -82,9 +82,11 @@ class VeloxTestSettings extends BackendTestSettings {
-    .exclude("FAILED_EXECUTE_UDF: execute user defined function")
+    // JDBC test
     .exclude("UNRECOGNIZED_SQL_TYPE: unrecognized SQL type -100")
+    // gluten throws different exception
```
Can you help investigate why the different exception occurs? And does it expect ?

ayushi-agarwal replied:
In spark, spark exception is thrown but in gluten in subquery transformer illegal state exception is thrown but the message is same so I have replicated the test in gluten side suite.

JkSelf commented on gluten-ut/spark34/src/test/scala/io/glutenproject/utils/velox/VeloxTestSettings.scala:
```diff
@@ -1156,10 +1160,10 @@ class VeloxTestSettings extends BackendTestSettings {
     .exclude("SPARK-33084: Add jar support Ivy URI in SQL")
     .exclude("SPARK-33084: Add jar support Ivy URI in SQL -- jar contains udf class")
-    // ReaderFactory is not registered for format orc.
+    // overflow support needed in velox
```
Can you help to investigate why 3.3 support the overflow and 3.4 not?

ayushi-agarwal replied:
This test was not present in 3.3 and this test is part of new feature added in spark 3.4

JkSelf commented on gluten-ut/spark34/src/test/scala/io/glutenproject/utils/velox/VeloxTestSettings.scala:
```diff
@@ -689,8 +693,10 @@ class VeloxTestSettings extends BackendTestSettings {
+    // add support in native reader
     .exclude("SPARK-41096: FIXED_LEN_BYTE_ARRAY support")
     .exclude("SPARK-40128 read DELTA_LENGTH_BYTE_ARRAY encoded strings")
+    // Timestamp is read as INT96
```
The spark 34 test is copied from spark 33. Could you please help investigate why the test works in Spark 33 but doesn't work in Spark 34?

ayushi-agarwal replied:
This test is also a new one in spark 3.4 and there already exists a excluded test in this file for the same reason. Once int64 support gets added these tests can be included.

JkSelf commented on gluten-ut/spark34/src/test/scala/io/glutenproject/utils/velox/VeloxTestSettings.scala:
```diff
@@ -82,9 +82,11 @@ class VeloxTestSettings extends BackendTestSettings {
     .exclude(
       "INCONSISTENT_BEHAVIOR_CROSS_VERSION: compatibility with Spark 2.4/3.2 in reading/writing dates")
-    .exclude("FAILED_EXECUTE_UDF: execute user defined function")
+    // JDBC test
```
Can you provide a detailed explanation as to why this suite needs to be excluded?

ayushi-agarwal replied:
This test needs to be excluded because it is jdbc related test which uses jdbc driver class which is not needed in gluten.This was also a new test in 3.4

ayushi-agarwal commented:
@JkSelf Thank you for reviewing. I have addressed your comments. There are some tests which were not present in 3.3 and got added in spark 3.4 as part of new features. There are work items like support for int64 and for others like new encodings or overflow new issues can be opened to provide a support for them in gluten/velox stack.

JkSelf reviewed (approved):
@ayushi-agarwal Thanks for your work. LGTM.
